### DISCUSSION TITLE
Is there an alternative to Discord (e.g. Matrix/IRC) for the Nushell community?

### DISCUSSION THREAD

**Discussion by squeaktoy** [nushell/nushell#11644] [General]
The fact the shell's welcome message demands I join the Discord guild makes me feel uneasy. I quit using Discord a long time ago because of its proprietary and defect-by-design nature. I don't understand why any open source project would even manifest its community on Discord. Let alone a shell. Discord doesn't run inside of a text shell last time I looked, but instead seems to require a web browser and a phone number, which is pretty bad because then nushell can't even be used to bootstrap access to the main community space. I do know an open source VR game (Overte) that has a bridge to Discord, considering that Discord is pretty much the main place to attract an audience for a social VR game, but for a Unix-like shell I heavily doubt Discord is the right place to attract an audience. It does block me from getting involved into it at least, considering I don't feel comfortable using a defect-by-design service like Discord.
Is there a Matrix bridge to the Discord guild, similar to how Overte operates, or an IRC channel just like how Odilia (the screen reader) operates?

**Comment by fdncred** [comment]
You've made your point that you don't like Discord but `Please join our Discord community at https://discord.gg/NtAbbGn` is not a demand. It's not very friendly to be so rude.

Last I checked there was a matrix group but I'm not sure how active it was. If you searched our issues, you would've seen this https://github.com/nushell/nushell/issues/9367. 

I've tried this out before. Not a lot of people here though.
https://app.element.io/#/room/#nushell:matrix.org

**Reply by squeaktoy** [reply]
It was never my intention to be rude. Don't assume malicious intent. The reason I would think it's a demand is because of the word `Please`, because people typically say please if they really want you to do something. I'm not sure if that's a thing they do in my country, or if there's some cultural difference in how the word please is used in your region and mine, but here it's used to demand something in a polite manner (although impolite is also possible).

**Reply by fdncred** [reply]
Using the word please is a request for you to do something if you so choose, and that's how we mean it. A demand would be "Join Discord, Now!". Agree that there could be cultural differences and that's why I took the time to explain.

**Reply by seungjin** [reply]
I am joining there.

**Reply by halloleo** [reply]
@squeaktoy _Regarding your feelings about the word "please"_:  It _can_ be interpreted as a demand, but this is very much dependend on quite culture and language. 

I think in English it is most of the time a friendly ask for something. Otherwise, in German (my mother tongue) "Bitte" (German for "Please") indeed can be - spoken in the right, stern voice - a strong demand.

In this context, I think, we should assume best intentions and see it as a friendly invite!

_Regarding your opinion about Discord_: I concur. I never liked Discord - but that might just show my age...

But we have the GitHub Discussionb here! They are pretty good! 😄

**Reply by squeaktoy** [reply]
GitHub is also gate keeped similarly to Discord, but not nearly as aggressive. It took me a year to make this account because GitHub's captcha was broken for a long time in 2023. I think open source projects and their communities should be accessible over open and federated or P2P protocols.

**Comment by crabdancing** [comment]
While it's definitely not intended to be a demand, I think it seems reasonable to add an official Matrix alternative to Discord.

While I do begrudgingly use Discord, it becomes more irritating to use every year, and definitely doesn't respect its users -- and other FOSS projects I've seen already do use Matrix for their primary out-of-SCM communities. It also looks like Matrix has an official Discord bridge in their Github repos: https://github.com/matrix-org/matrix-appservice-discord

It would require some cooperation with whoever is managing the Discord guild, as the bot has to be specifically allowed by whoever owns the guild IIRC. But this would be a great alternative to have for people. I'm all for it! :)
